xds/internal/balancer/outlierdetection/balancer_test.go

```diff
@@ -270,20 +297,1551 @@ func (lbc *LBConfig) Equal(lbc2 *LBConfig) bool {
+	}
+	dur := d.(time.Duration)
+	// The configured duration should be 8 seconds - what the balancer was
+	// configured with.
+	if dur.Seconds() != 8 {
+		t.Fatalf("configured duration should have been 8 seconds to start timer")
+	}
+
+	defer func(n func() time.Time) {
+		now = n
+	}(now)
+	now = func() time.Time {
+		return time.Now().Add(time.Second * 5)
+	}
+
+	// UpdateClientConnState with an interval of 9 seconds. Due to 5 seconds
+	// already passing (from overridden time.Now function), this should start an
+	// interval timer of ~4 seconds.
+	od.UpdateClientConnState(balancer.ClientConnState{
+		BalancerConfig: &LBConfig{
+			Interval:           9 * time.Second,
+			BaseEjectionTime:   30 * time.Second,
+			MaxEjectionTime:    300 * time.Second,
+			MaxEjectionPercent: 10,
+			SuccessRateEjection: &SuccessRateEjection{
+				StdevFactor:           1900,
+				EnforcementPercentage: 100,
+				MinimumHosts:          5,
+				RequestVolume:         100,
+			},
+			FailurePercentageEjection: &FailurePercentageEjection{
+				Threshold:             85,
+				EnforcementPercentage: 5,
+				MinimumHosts:          5,
+				RequestVolume:         50,
+			},
+			ChildPolicy: &internalserviceconfig.BalancerConfig{
+				Name:   tcibname,
+				Config: balancerConfig{},
+			},
+		},
+	})
+
+	d, err = durationChan.Receive(ctx)
+	if err != nil {
+		t.Fatalf("Error receiving duration from afterFunc() call: %v", err)
+	}
+	dur = d.(time.Duration)
+	if dur.Seconds() < 3.5 || 4.5 < dur.Seconds() {
+		t.Fatalf("configured duration should have been around 4 seconds to start timer")
+	}
+
+	// UpdateClientConnState with a no-op config. This shouldn't configure the
+	// interval timer at all due to it being a no-op.
+	od.UpdateClientConnState(balancer.ClientConnState{
+		BalancerConfig: &LBConfig{
+			Interval: 10 * time.Second,
+			ChildPolicy: &internalserviceconfig.BalancerConfig{
+				Name:   tcibname,
+				Config: balancerConfig{},
+			},
+		},
+	})
+
+	ctx, cancel = context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	// No timer should have been started.
+	sCtx, cancel := context.WithTimeout(context.Background(), defaultTestShortTimeout)
+	defer cancel()
+	_, err = durationChan.Receive(sCtx)
+	if err == nil {
+		t.Fatal("No timer should have started.")
+	}
+}
+
+// TestEjectUnejectSuccessRate tests the functionality of the interval timer
+// algorithm of ejecting/unejecting SubConns when configured with
+// SuccessRateEjection. It also tests a desired invariant of a SubConnWrapper
+// being ejected or unejected, which is to either forward or not forward SubConn
+// updates received from grpc.
+func (s) TestEjectUnejectSuccessRate(t *testing.T) {
+	scsCh := testutils.NewChannel()
+	var scw1, scw2, scw3 balancer.SubConn
+	var err error
+	stub.Register(tcibname, stub.BalancerFuncs{
+		UpdateClientConnState: func(bd *stub.BalancerData, _ balancer.ClientConnState) error {
+			scw1, err = bd.ClientConn.NewSubConn([]resolver.Address{
```

Comment:
Same comment here. Simply call `NewSubConn` on the addresses received instead of hardcoding.